ProcessLib/ThermoMechanics/ThermoMechanicsProcess.cpp

```diff
@@ -134,6 +155,65 @@ void ThermoMechanicsProcess<DisplacementDim>::initializeConcreteProcess(
             // by location order is needed for output
             NumLib::ComponentOrder::BY_LOCATION));
 
+    // Vector of mesh subsets.
+    std::vector<MeshLib::MeshSubset> all_mesh_subsets;
+
+    // Vector of the number of variable components
+    std::vector<int> vec_var_n_components;
+    if (_use_monolithic_scheme)
+    {
+        // Collect the mesh subsets in a vector for each variables' components.
+        for (ProcessVariable const& pv : _process_variables[0])
+        {
+            std::generate_n(std::back_inserter(all_mesh_subsets),
+                            pv.getNumberOfComponents(),
+                            [&]() { return *_mesh_subset_all_nodes; });
+        }
+
+        // Create a vector of the number of variable components
+        for (ProcessVariable const& pv : _process_variables[0])
+        {
+            vec_var_n_components.push_back(pv.getNumberOfComponents());
+        }
+    }
+    else  // for staggered scheme
+    {
+        // Collect the mesh subsets in a vector for each variable components for
+        // the mechanical process.
+        std::generate_n(std::back_inserter(all_mesh_subsets),
+                        _process_variables[_mechanics_process_id][0]
+                            .get()
+                            .getNumberOfComponents(),
+                        [&]() { return *_mesh_subset_all_nodes; });
+
+        // Create a vector of the number of variable components.
+        vec_var_n_components.push_back(
```

Comment:
For discussion: Why only one push_back for mechanics and no for T?

Reply:
Process::constructDofTable() handles the cases that all processes of a problem has the same number of primary variable (or the same DOF per node in FEM), which means all processes use a single  DOF table.

For TM, T and M have different DOF tables. Therefore  Process::constructDofTable() is overloaded in   ThermoMechanicsProcess.
DOF table:
T: _local_to_global_index_map_single_component, a member of ThermoMechanicsProcess.
M: _local_to_global_index_map, a member of Process.

Reply:
Process::constructDofTable() can be refactored later on to reduce code duplication in  the overloaded member of it in HM, PM. The refactored  Process::constructDofTable() can initialize Process::_local_to_global_index_map for all process, and the overloaded constructDofTable() can call Process::constructDofTable() and initialize the specific DOF table   of the derived classes.